scm-plugin/src/main/resources/archetype-resources/src/main/java/HelloWorldSCM.java

```diff
@@ -0,0 +1,61 @@
+package $package;
+
+import java.io.File;
+import java.io.IOException;
+
+import hudson.Extension;
+import hudson.FilePath;
+import hudson.Launcher;
+import hudson.model.Job;
+import hudson.model.Run;
+import hudson.model.TaskListener;
+import hudson.scm.ChangeLogParser;
+import hudson.scm.PollingResult;
+import hudson.scm.SCM;
+import hudson.scm.SCMDescriptor;
+import hudson.scm.SCMRevisionState;
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.SCMRevision;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class HelloWorldSCM extends SCM {
+    HelloWorldSCM(SCMHead head, SCMRevision revision) {
+
+    }
+
+    @Override
+    public ChangeLogParser createChangeLogParser() {
+        return new HelloWorldSCMChangeLogParser();
+    }
+
+    @Override
+    public PollingResult compareRemoteRevisionWith(@Nonnull Job<?, ?> project, @Nullable Launcher launcher, @Nullable FilePath workspace, @Nonnull TaskListener listener, @Nonnull SCMRevisionState baseline) throws IOException, InterruptedException {
+        return null;
```

Comment:
I do not think `null` is a legitimate return value. Probably better to throw `NotImplementedException` or something for critical methods like this.